src/main/kotlin/org/rust/ide/actions/RustFmtFileAction.kt

```diff
@@ -0,0 +1,60 @@
+package org.rust.ide.actions
+
+import com.intellij.execution.ExecutionException
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.module.ModuleUtilCore
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.util.PathUtil
+import org.rust.cargo.toolchain.RustToolchain
+import org.rust.ide.notifications.showBalloon
+import org.rust.lang.RustFileType
+
+class RustFmtFileAction : DumbAwareAction() {
+    override fun update(e: AnActionEvent?) {
+        super.update(e);
+        e ?: return
+        val project = e.project
+        val file = e.getData(CommonDataKeys.VIRTUAL_FILE)
+        val editor = e.getData(CommonDataKeys.EDITOR)
+
+        if (project == null || file == null || !file.isInLocalFileSystem() || !isAvailableOnFile(file) || editor == null) {
+            e.presentation.isEnabled = false
+            return
+        }
+        e.presentation.isEnabled = true
+    }
+
+    protected fun isAvailableOnFile(file: VirtualFile): Boolean {
+        return file.fileType === RustFileType
+    }
```

Comment:
There is `isRustFile` extension for VirtualFiles. So I think this helper method can be omitted.